velox/common/file/FileSystems.h

```diff
@@ -45,6 +45,10 @@ class FileSystem {
 
   // Deletes the file at 'path'. Throws on error.
   virtual void remove(std::string_view path) = 0;
+  // Returns true if the file exists.
+  virtual bool exist(std::string_view path) = 0;
```

Comment:
exist -> exists with an 's' at the end for singular since we are asking whether a single path exists or not.

It would be nice to document that exists and list methods are for testing only and therefore optional to implement. I'd also clarify that currently only local file system implements these methods.

Reply:
Should we make these `virtual functions` and not `pure virtual functions` since these are optional?